pkg/configurator/types.go

```diff
@@ -89,4 +89,17 @@ type Configurator interface {
 
 	// GetProxyResources returns the `Resources` configured for proxies, if any
 	GetProxyResources() corev1.ResourceRequirements
+
+	// GetInboundExternalAuthConfig returns the External Authentication configuration for incoming traffic, if any
+	GetInboundExternalAuthConfig() ExternAuthConfig
+}
+
+// ExternAuthConfig implements a generic subset of External Authz to configure external authorization in envoy's format
+type ExternAuthConfig struct {
+	Enable           bool
+	Address          string
+	Port             uint16
+	StatPrefix       string
+	AuthzTimeout     time.Duration
+	FailureModeAllow bool
```

Comment:
I'd create a separate pkg for this `pkg/auth` and put this definition there so it can be reused by configurator and XDS.

Reply:
So, Q for you. Here I added the extra structure in config, but we could have just propagated the `v1alpha.ExternalAuthzSpec` that comes directly from meshconfig (after all, they both hold the exact same things).
Though this structure does not bring anything new, I thought we might want a different structure to represent segment what inner parts of OSM do not interact directly with MeshSpec, and only configurator does.

wdyt? would you skip this intermediate structure?

Reply:
That's a good question. It depends on if you want to tie API definitions to internal implementations. For example, if there is ever a case where an internal struct does not need to be the same as the API Go struct, or if the internal struct can be extended with additional metadata, then it makes sense to keep it separate so you don't have to change the API (new API version rollout). Updating a CRD in theory should be a version update once users start using the API.

However if you don't forsee such a scenario, it makes sense to avoid the intermediate struct mapping.